### PR TITLE
[libtorrent] Update to latest 1.2.x commit 

### PR DESCRIPTION
**Pull request by FranciscoPombal**
- Update to 1.2.13 plus some additional fixes (allows dropping existing
patches)
- Build with libiconv by default on all platforms
- Fix deprecated usage of `vcpkg_check_features` without the `FEATURES`
keyword
- Migrate to vcpkg.json manifest files

---

 [libtorrent] Update versioning information

---

Previously: https://github.com/microsoft/vcpkg/pull/15620 and https://github.com/microsoft/vcpkg/pull/14755.

1.2.14 may take a while to come, and this is a good opportunity to get rid of all patches and migrate to manifest files. Hopefully this will make future work of getting the python feature to work on Windows easier.

Tested on`x64-linux`,  `x64-windows`, and `x64-windows-static`.

### CODE REVIEW

File: versions/l-/libtorrent.json [hunk 1]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "cefe40e87e8a190462a748b1a230cedac39e8395",
+      "version-string": "1.2.13-b799640",
```
Comment (JonLiu1993) [inline versions/l-/libtorrent.json:5]: @FranciscoPombal ,we should  change version-string to version

File: ports/libtorrent/vcpkg.json [hunk 2]
```diff
@@ -0,0 +1,53 @@
+{
+  "name": "libtorrent",
+  "version-string": "1.2.13-b799640",
```
Comment (JonLiu1993) [inline ports/libtorrent/vcpkg.json:3]: we should change version-string to version

**Comment by JonLiu1993** [comment]
@FranciscoPombal ，Thanks for your contribute

File: versions/l-/libtorrent.json (hunk 1 above)
Comment (FranciscoPombal) [reply versions/l-/libtorrent.json]: fixed.

File: ports/libtorrent/vcpkg.json (hunk 2 above)
Comment (FranciscoPombal) [reply ports/libtorrent/vcpkg.json]: fixed.

**Comment by FranciscoPombal** [comment]
@JonLiu1993 Looks like we have to use "version-string":

```
$ (vcpkg.json): 'version' text was not a relaxed version:
Error: String `1.2.13-b799640` must only contain dot-separated numeric values without leading zeroes.
```

**Review by ytbigvinny27 — APPROVED** [review]

**Review by ytbigvinny27 — APPROVED** [review]

**Review by JonLiu1993 — APPROVED** [review]

**Comment by FranciscoPombal** [comment]
@JonLiu1993 Anything preventing this from being merged?

**Comment by JonLiu1993** [comment]
@FranciscoPombal ,Don’t worry, soon this pr will be merged with other prs.

**Comment by strega-nil-ms** [comment]
Thanks @FranciscoPombal :)

**Pull request merged**